syntax = "proto3";

package core.platform.v1;

import "google/protobuf/timestamp.proto";
import "platform/v1/entities.proto";

enum EventType {
  EVENT_TYPE_UNSPECIFIED = 0;
  STUDENT_REGISTERED = 1;
  VACANCY_POSTED = 2;
  APPLICATION_CREATED = 3;
  APPLICATION_STATUS_CHANGED = 4;
}

message BaseEvent {
  string event_id = 1;
  EventType type = 2;
  google.protobuf.Timestamp occurred_at = 3;
}

message ApplicationCreatedEvent {
  BaseEvent base = 1;
  int64 application_id = 2;
  int64 student_id = 3;
  int64 vacancy_id = 4;
}

message ApplicationStatusChangedEvent {
  BaseEvent base = 1;
  int64 application_id = 2;
  ApplicationStatus old_status = 3;
  ApplicationStatus new_status = 4;
}
